src/util/file_reader.ts

```diff
@@ -120,37 +120,56 @@ async function collectLocalPathBehaviors(
     return [];
   }
 
-  const stat = await fsp.stat(resolvedPath);
+  const behaviors: FileSources = [];
 
-  if (stat.isFile() && ALLOWED_EXTS.includes(path.extname(resolvedPath))) {
-    const contents = await fsp.readFile(resolvedPath);
-    return [
-      {
-        path: resolvedPath,
-        contents: `/* src: ${resolvedPath} */\n\n${contents}`,
-      },
-    ];
-  }
+  try {
+    const stat = await fsp.stat(resolvedPath);
+
+    if (stat.isFile() && ALLOWED_EXTS.includes(path.extname(resolvedPath))) {
+      const contents = await fsp.readFile(resolvedPath);
+      return [
+        {
+          path: resolvedPath,
+          contents: `/* src: ${resolvedPath} */\n\n${contents}`,
+        },
+      ];
+    }
 
-  const behaviors: FileSources = [];
+    const isDir = stat.isDirectory();
 
-  const isDir = stat.isDirectory();
+    if (!isDir && depth === 0) {
+      logger.warn(
+        "The provided path is not a .js file or directory",
+        { path: resolvedPath },
+        "behavior",
+      );
+    }
 
-  if (!isDir && depth === 0) {
-    logger.warn(
-      "The provided path is not a .js file or directory",
-      { path: resolvedPath },
+    if (isDir) {
+      const files = await fsp.readdir(resolvedPath);
+      for (const file of files) {
+        const filePath = path.join(resolvedPath, file);
+        const newBehaviors = await collectLocalPathBehaviors(
+          filePath,
+          depth + 1,
+        );
+        behaviors.push(...newBehaviors);
+      }
+    }
+  } catch (e) {
+    logger.fatal(
+      "Error fetching local custom behaviors",
+      { path: resolvedPath, error: e },
       "behavior",
     );
   }
 
-  if (isDir) {
-    const files = await fsp.readdir(resolvedPath);
-    for (const file of files) {
-      const filePath = path.join(resolvedPath, file);
-      const newBehaviors = await collectLocalPathBehaviors(filePath, depth + 1);
-      behaviors.push(...newBehaviors);
-    }
+  if (!behaviors && depth === 0) {
```

Comment:
Hm, does this mean that specifying an empty directory will lead to an error? I think that could be allowed, as long as directory exists?

Reply:
(Pointing out Git repo here because we first clone the git repo and then collect behaviors from the cloned repo via this codepath)

Reply:
Yeah, just thinking if maybe should allow having a placeholder directory that things will be added to later.. I guess could always add a sample behavior or something like that..

Reply:
Yeah if we go that way I'd suggest we do add a sample behavior. Otherwise we might be setting people up for silent errors?

Reply:
I'll leave it up to your judgment though